rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+# Networked Dot Storage
+
+> or what does decentralziation even mean?
+
+## Authors
+
+- [Hannah Howard], [Protocol Labs]
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Abstract
+
+This RFC puts forward a proposal for the broad parameters of "decentralization" in W3S. It defines both the goals for "decentralization" and identifies where resources should be invested first on the path to "decentralization".
+
+## Non-goals
+
+1. Addressing anything related to tokens
+2. W3S corporate structure or nucleation plans
+
+## Background
+
+The current state of the world for web3.storage is as follows:
+
+- Web3.storage UCAN protocols are open and well specified. 
+
+- We've now created a version of these protocols that can be run from any machine-- so anyone can spin up theoretically own version of our service.
+
+- Seperately Web3.storage backs up data to Filecoin, via open protocols.
+
+- We are actively moving towards supporting running on top of underlying storage mechanisms that lack rich event functionality (a.k.a. bucket events)
+
+## Why Go Further
+
+Today, someone who is not the web3.storage team could spin up their own version of web3.storage, using the open source software we've published and the protocols we've specified. No one has done this in practice, but we are increasingly close to the point where it's plausable. Is this "sufficient decentralization"?
+
+One proposed path forward is to not pursue "decentralization" beyond trying to make other nodes a reality, perhaps through incentivization, and leaning heavily into product improvements that differentiate our platform. We could aim for "S3 if AWS invested in open source S3 implementations and made it content addressed".
+
+This is not the path we are going to pursue. It feels important to say that up front.
+
+But before we go deeper into where we are headed, it's worth looking at the implied dichotomy between "decentralization" and product features. That this dichotomy is almost reflexively referred to likely results from failing to appropriately frame "decentralization" in terms of product features. So let's start by framing the product features we want to unlock through decentralization.
+
+## The Network We Want
+
+### No Data Loss
+
+1. Users should be able to specify how many unique copies of their data they want backed up, and the W3S network should handle insuring that many copies remain backed up as long as the user pay to store in W3S.
```

Comment:
some food for thought: more than replication, I have always liked the idea of allowing user custom region, or even selecting providers with different SLAs. For instance, one node running this service could host things in S3 EU Region, with greater SLA characteristics. Storage costs SHOULD not be tied to what web3.storage says, but what each provider offers to enable a more open market

Reply:
This is a nuance, but I can't help but wonder if making users choose number of unique copies is the right approach, my guess is users want some resilience, perhaps geographic distribution and other forms of assurances that data availability is adequate. Trying to figure out number of replicas seems several steps disconnected from those needs and also would remove some flexibility on the implementer side.

Reply:
Yeah honestly I think this should be a much more nuanced thing - imho different w3up providers should be able to offer different levels of "quality of service" on a bunch of different axes (latency, resilience, bandwidth, etc) as well different levels of legal/social guarantees (say, "we will comply with GDPR" or "we will never host data in Iran/China/Russia/USA/etc") - we could then craft different "default" user experiences for people that connected them with providers that met their needs, and once they are in the ecosystem they wouldn't necessarily need to maintain a relationship with us if they don't want to - I think this could go a long way toward balancing "we want to build a defensible business" and "we want to minimize our role as a central point of failure or point of centralization for our users" which imho are in dramatic tension for us...

Reply:
oh also I think this might even open up space for people to make decisions about storage providers based on real-world social factors, which is imho what is needed to support "private data" or PII - this is honestly the market segment I am most excited about

Reply:
> This is a nuance, but I can't help but wonder if making users choose number of unique copies is the right approach, my guess is users want some resilience, perhaps geographic distribution and other forms of assurances that data availability is adequate. Trying to figure out number of replicas seems several steps disconnected from those needs and also would remove some flexibility on the implementer side.

Yes, +1. In the same way that I do not care how many replicas AWS stores of the data I have in S3. It should be adequete, so that the data is not lost.

When we talk about geographical distribution, my suspicion is that we're mostly thinking about retrieving the data quickly, which is what the CDN side of the product should be oriented around, not the storage side.